### PR TITLE
Adding python3-subscription-manager-rhsm for pulp builds

### PR DESCRIPTION
**Pull request by jlsherrill**
The goal of this is to build python3-subscription-manager-rhsm
in order to use certguard with rhsm cert protection.  I think this is okay given:
* this is only needed until we move to el8
* it wont need to stay in lock-step with the subscription-manager version installed
* its unlikely to need updates

### CODE REVIEW

File: package_manifest.yaml [hunk 1]
```diff
@@ -1351,6 +1351,7 @@ pulpcore_packages:
     python-xlwt: {}
     python-yarl: {}
     python-zipp: {}
+    subscription-manager: {}
```
Comment (ehelms) [inline package_manifest.yaml:1354]: Add

```
    koji_tags:
      - name: katello-pulpcore-nightly-el7
        dist: '.el7'
```

**Comment by ehelms** [comment]
You are gonna have to introduce a blacklist for https://github.com/theforeman/foreman-packaging/pull/5363/files#diff-150be58e1a8579150d5a3803c4f60d6cR1246 to stop the EL8 build as well.

**Comment by jlsherrill** [comment]
This is now green!  @evgeni i did have to hack up the spec files quite a bit, mostly to get repoclsoure to pass since we only care about one of the subpackages of rhsm.  Any objections?  If this was a longer term solution i'd not go this route, but this buys us time to get to el8

**Comment by jlsherrill** [comment]
and i think that the python3-subscrption-manager-rhsm package should be installed via the installer, and not via a dependency of the certguard plugin

**Comment by evgeni** [comment]
Fwiw, py-iniparse in el8 does not build py2, so it might be cleaner to use that spec as a base? https://koji.mbox.centos.org/koji/buildinfo?buildID=430

That said, I don't care too much as long as these hacks are self contained and vanish at some point where we stop supporting el7 installs of pulpcore (right @ehelms, that will be soon?!).

File: packages/pulpcore/subscription-manager/subscription-manager.spec [hunk 2]
```diff
@@ -0,0 +1,5935 @@
...
+Group: Development/Libraries
+%endif
+
+%if %use_m2crypto
+Requires: %{?suse_version:python-m2crypto} %{!?suse_version:m2crypto}
+%endif
+%if 0%{?suse_version} >= 1500
+Requires:  %{py_package_prefix}-python-dateutil
+%else
+Requires: python36-dateutil
```
Comment (evgeni) [inline packages/pulpcore/subscription-manager/subscription-manager.spec:558]: That's coming from EPEL, is that correct?

File: packages/pulpcore/subscription-manager/subscription-manager.spec [hunk 3]
```diff
@@ -0,0 +1,5935 @@
...
+%endif
+Requires: python36-iniparse
```
Comment (evgeni) [inline packages/pulpcore/subscription-manager/subscription-manager.spec:560]: Technically, this should be `python3-iniparse`, but your package also provides the 36 name.

File: packages/pulpcore/subscription-manager/subscription-manager.spec [hunk 4]
```diff
@@ -0,0 +1,5935 @@
...
+Requires: %{py_package_prefix}-six
+Requires: subscription-manager-rhsm-certificates
+# Required by Fedora packaging guidelines
+%{?python_provide:%python_provide %{py_package_prefix}-rhsm}
+%if %{with python3}
+Requires: python3-rpm
+Provides: python3-rhsm = %{version}-%{release}
+Obsoletes: python3-rhsm <= 1.20.3-1
+Provides: python-rhsm = %{version}-%{release}
+Obsoletes: python-rhsm <= 1.20.3-1
```
Comment (evgeni) [inline packages/pulpcore/subscription-manager/subscription-manager.spec:571]: I don't think we should provide/obsolete these, given that on EL7 that are py2 versions.

File: packages/pulpcore/subscription-manager/subscription-manager.spec (hunk 2 above)
Comment (jlsherrill) [reply packages/pulpcore/subscription-manager/subscription-manager.spec:558]: correct!

File: packages/pulpcore/subscription-manager/subscription-manager.spec (hunk 4 above)
Comment (jlsherrill) [reply packages/pulpcore/subscription-manager/subscription-manager.spec]: fixed!

File: packages/pulpcore/subscription-manager/subscription-manager.spec (hunk 3 above)
Comment (jlsherrill) [reply packages/pulpcore/subscription-manager/subscription-manager.spec]: fixed

**Comment by jlsherrill** [comment]
i think i've addressed all the comments/questions

**Pull request merged**